lte/gateway/c/core/oai/tasks/ngap/ngap_amf_nas_procedures.c

```diff
@@ -1211,36 +1211,33 @@ int ngap_amf_nas_pdusession_resource_setup_stream(
         &(session_item->PDU_Session_Resource_Setup_Request_Transfer);
 
     /*tx_out*/
-    Ngap_PDUSessionResourceSetupRequestTransfer_t*
-        pduSessionResourceSetupRequestTransferIEs =
-            (Ngap_PDUSessionResourceSetupRequestTransfer_t*)calloc(
-                1, sizeof(Ngap_PDUSessionResourceSetupRequestTransfer_t));
+    Ngap_PDUSessionResourceSetupRequestTransfer_t msg;
+    memset(&msg, 0, sizeof(Ngap_PDUSessionResourceSetupRequestTransfer_t));
 
     ngap_fill_pdu_session_resource_setup_request_transfer(
-        amf_pdu_ses_setup_transfer_req,
-        pduSessionResourceSetupRequestTransferIEs);
+        amf_pdu_ses_setup_transfer_req, &msg);
     uint32_t buffer_size = 1024;
     char* buffer = (char*)calloc(1, buffer_size);
 
     asn_enc_rval_t er = aper_encode_to_buffer(
-        &asn_DEF_Ngap_PDUSessionResourceSetupRequestTransfer, NULL,
-        pduSessionResourceSetupRequestTransferIEs, buffer, buffer_size);
+        &asn_DEF_Ngap_PDUSessionResourceSetupRequestTransfer, NULL, &msg,
+        buffer, buffer_size);
 
     if (er.encoded <= 0) {
       OAILOG_ERROR(LOG_NGAP, "PDU Session Resource Request IE encode error \n");
       OAILOG_FUNC_RETURN(LOG_NGAP, RETURNerror);
     }
 
     asn_fprint(stderr, &asn_DEF_Ngap_PDUSessionResourceSetupRequestTransfer,
-               pduSessionResourceSetupRequestTransferIEs);
+               &msg);
 
-    bstring transfer = blk2bstr(buffer, er.encoded);
+    bstring transfer = blk2bstr(buffer, ((er.encoded + 7) >> 3));
```

Comment:
Add comment to explain the purpose of operation, ((er.encoded + 7) >> 3)

Reply:
Comment addressed.

Reply:
@priya-wavelabs Could you please explain why we use 7 and 3. Can we deduce the length from buffer based on number of bytes encoded